test/core/end2end/fixtures/http_proxy_fixture.c

```diff
@@ -352,6 +394,30 @@ static void on_read_request_done(grpc_exec_ctx* exec_ctx, void* arg,
     GRPC_ERROR_UNREF(error);
     return;
   }
+  // If proxy auth is being used, check if the header is present and as expected
+  if(grpc_channel_args_find(
+      conn->proxy->channel_args,
+      GRPC_END2END_HTTP_PROXY_TEST_CONNECT_AUTH_PRESENT) != NULL) {
+    bool auth_header_found = false;
+    for(size_t i = 0; i < conn->http_request.hdr_count; i++) {
+      if(strcmp(conn->http_request.hdrs[i].key, "Proxy-Authorization") == 0) {
+        if(!proxy_auth_header_matches(
+            exec_ctx, conn->http_request.hdrs[i].value)) {
+          break;
+        }
+        auth_header_found = true;
+        break;
+      }
+    }
+    if(!auth_header_found) {
+      const char *msg = "HTTP Connect could not verify authentication";
+      error = GRPC_ERROR_CREATE_FROM_COPIED_STRING(msg);
```

Comment:
Can use `GRPC_ERROR_CREATE_FROM_STATIC_STRING()` here, since this is a string literal.